presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -91,15 +91,15 @@ public SystemSessionProperties(
                         featuresConfig.isOptimizeHashGeneration(),
                         false),
                 booleanSessionProperty(
-                        DISTRIBUTED_JOIN,
-                        "Use a distributed join instead of a broadcast join",
-                        featuresConfig.isDistributedJoinsEnabled(),
-                        false),
-                booleanSessionProperty(
                         DISTRIBUTED_INDEX_JOIN,
                         "Distribute index joins on join keys instead of executing inline",
                         featuresConfig.isDistributedIndexJoinsEnabled(),
                         false),
+                stringSessionProperty(
+                        JOIN_DISTRIBUTION_TYPE,
+                        "What join method to use. Options are partitioned, replicated, or automatic",
```

Comment:
What was your final decision to call this, partitioned or repartitioned? I saw in docs both the names being used. Lets pick one